action/rlp_tx.go

```diff
@@ -83,3 +82,68 @@ func DecodeRawTx(rawData string, chainID uint32) (tx *types.Transaction, sig []b
 	pubkey, err = crypto.RecoverPubkey(rawHash[:], sig)
 	return
 }
+
+// DecodeEtherTx decodes raw data string into eth tx
+func DecodeEtherTx(rawData string) (*types.Transaction, iotextypes.Encoding, []byte, crypto.PublicKey, error) {
```

Comment:
we may decouple the generation of `tx` and `encoding, sig, publickey` into `func DecodeEthTx(rawData string) (*types.Transaction, error)` and `func ExtractXXX(tx) (iotextypes.Encoding, []byte, crypto.PublicKey, error)`
```

Reply:
we could, but just as you commented below, it is only called 1 time in `api` package. If splitting to 2 parts can make them re-use more than 1 times, then it's definitely worth it

Reply:
`DecodeEtherTx` is not clearly describe what it is doing. Legacy problem cannot be used as excuse to latest implementation especially that the legacy function is not used.

Reply:
updated in latest commit